internal/util/observability/funccall.go

```diff
@@ -36,31 +40,39 @@ type funcCall struct {
 // The only valid way to use FuncCall is:
 //
 //	func foo(ctx context.Context) {
-//	    defer FuncCall(ctx)()
-//	    // ...
+//		ctx, leave := FuncCall(ctx)
+//		defer leave()
+//
+//		// ...
+//
+// For OpenTelemetry tracing, FuncCall creates a new span for the function call FIXME.
 //
 // For the Go execution tracer, FuncCall creates a new region for the function call
 // and attaches it to the task in the context (or background task).
-func FuncCall(ctx context.Context) func() {
+func FuncCall(ctx context.Context) (context.Context, func()) {
```

Comment:
Doesn't this end up limiting too much? Like, in the past I've found adding events useful and then we'd need to have another way to do so.

Reply:
Maybe. It is a draft, after all